app/liquid_tags/liquid_tag_base.rb

```diff
@@ -3,11 +3,39 @@ def self.script
     ""
   end
 
+  def initialize(_tag_name, _content, _parse_context)
+    super
+    validate_contexts
+    # This check issues DB queries so we keep it as the last one
+    Pundit.authorize(
+      parse_context.partial_options[:user],
+      self,
+      :initialize?,
+      policy_class: LiquidTagPolicy,
+    )
+  end
+
   def finalize_html(input)
     input.gsub(/ {2,}/, "").
       gsub(/\n/m, " ").
       gsub(/>\n{1,}</m, "><").
       strip.
       html_safe
   end
+
+  private
+
+  def validate_contexts
+    return unless self.class.const_defined? "VALID_CONTEXTS"
```

Comment:
If you don't want to go through the class, you can use `defined?(VALID_CONTEXTS)`. It has a somewhat weird output (`nil` or `"constant"`), but works for truthiness checks.

Reply:
I couldn't get this to work 🤔. `defined?(VALID_CONTEXTS)` returned `nil` in the cases when I expected it not to.

Reply:
Nevermind, it's not important. It may not work as I thought for inherited constants for example 😄